pkg/secretless/entrypoint_test.go

```diff
@@ -0,0 +1,91 @@
+package secretless
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/cyberark/secretless-broker/internal/pkg/plugin"
+)
+
+var testManager = &plugin.Manager{}
+
```

Comment:
The `assertOutputMatches` func is too complex imo.  It also has too many responsibilities, and re-implements [assertions testify already has](https://godoc.org/github.com/stretchr/testify/assert#Regexp) (since we've officially landed on `testing` + `testify` assertions as our unit testing strategy, we should use them.

Breaking down the above in detail....

1. *Too complex* We've got a lot of code here, including the rather hacky hardcoded swapping of stdout, and even creating goroutines to deal with pipe io.  The test code seems more complex than the code it's testing, which is never a good thing.  From a design perspective, I'd much rather see us using dependency injection here if we're going to write tests against what's being printed on stdout. 
2. *Too many responsibilities* This code is a combination of regex assertion code, utility code for mocking stdout, and code for kicking off the application func under test (ie, `Start`).  A better approach is to use the testify assertions for the regex part, have dedicated utility code for the stdout mocking (or better yet, delete it altogether and use DI, per above), and kick off `Start` in the actual tests.

Reply:
I'll move to `testify` assertions since it proved better in the other code but the other points I'll raise in dev channel after lunch